proof/crefine/ARM_HYP/ArchMove_C.thy

```diff
@@ -634,6 +634,38 @@ crunches insertNewCap, Arch_createNewCaps, threadSet, Arch.createObject, setThre
    simp: unless_def updateObject_default_def crunch_simps
    ignore_del: preemptionPoint)
 
+(* FIXME MK: consider move *)
+lemma addrFromPPtr_mask[simplified ARM_HYP.pageBitsForSize_simps]:
+  "n \<le> pageBitsForSize ARMSuperSection
+   \<Longrightarrow> addrFromPPtr ptr && mask n = ptr && mask n"
+  apply (simp add: addrFromPPtr_def)
+  apply (prop_tac "pptrBaseOffset AND mask n = 0")
+   apply (rule mask_zero[OF is_aligned_weaken[OF pptrBaseOffset_aligned]], simp)
+  apply (simp flip: mask_eqs(8))
+  done
+
+(* FIXME MK: consider move *)
+lemma ptrFromPAddr_mask[simplified ARM_HYP.pageBitsForSize_simps]:
+  "n \<le> pageBitsForSize ARMSuperSection
+   \<Longrightarrow> ptrFromPAddr ptr && mask n = ptr && mask n"
+  apply (simp add: ptrFromPAddr_def)
+  apply (prop_tac "pptrBaseOffset AND mask n = 0")
+   apply (rule mask_zero[OF is_aligned_weaken[OF pptrBaseOffset_aligned]], simp)
+  apply (simp flip: mask_eqs(7))
+  done
+
+(* FIXME MK: this can be done as:
+   lemmas addrFromPPtr_mask_6 = addrFromPPtr_mask[where n=6, simplified]
+   but that doesn't give a sanity check of the n \<le> ... assumption  disappearing *)
+lemma addrFromPPtr_mask_6:
+  "addrFromPPtr ptr && mask 6 = ptr && mask 6"
+  by (rule addrFromPPtr_mask[where n=6, simplified])
```

Comment:
Same as the `addrFromPPtr_mask_5` I'd drop the FIXME here. Also for the following `ptrFromPAddr_mask_6`